core/Tracker/Visit.php

```diff
@@ -562,7 +562,18 @@ private function setIdVisitorForExistingVisit($valuesToUpdate)
             // Might update the idvisitor when it was forced or overwritten for this visit
             $valuesToUpdate['idvisitor'] = $this->request->getVisitorId(); 
         }
-        
+
+        if (TrackerConfig::getConfigValue('enable_userid_overwrites_visitorid')) {
```

Comment:
technically, this `if` might be no longer needed since we added above `if` in Matomo 4. However, the `if` above does not yet update `$this->visitProperties->setProperty('idvisitor', $binIdVisitor);` and I'm not sure why this is done and if it's needed so I added the entire `if` block